src/test/java/org/junitpioneer/jupiter/UtilsTest.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2015-2020 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junitpioneer.jupiter;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import java.util.HashSet;
+import java.util.Set;
+import java.util.function.BiConsumer;
+import java.util.function.BinaryOperator;
+import java.util.function.Supplier;
+import java.util.stream.Collector;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+@DisplayName("Utils")
+class UtilsTest {
+
+	@Nested
+	@DisplayName("to distinct set")
+	class ToDistinctSetTests {
+
+		Collector<Object, Set<Object>, Set<Object>> collector;
+
+		@BeforeEach
+		void setUp() throws Exception {
+			collector = Utils.distinctToSet();
+		}
+
+		@Test
+		@DisplayName("should use hash set")
+		void supplierShouldUseHashSet() throws Exception {
+			Supplier<Set<Object>> supplier = collector.supplier();
+
+			assertThat(supplier.get()).isInstanceOf(HashSet.class);
```

Comment:
Do you think we should really verify that it's a `HashSet` - any other implementation should be fine as well, right?

Reply:
Yes, another implementation would do the job too. I just thought it would make sense to specify the current implementation as part of the tests. Feel free to delete the test case.

Reply:
I'll do that then. :)